### PR TITLE
Introduce task deployment constraints 

### PR DESCRIPTION
**Pull request by f4tq**
- The etcd framework constraints are not applied to the task's it creates  -- etcd itself -- resulting in etcd running anywhere.
- This PR provides a way to confine tasks using constraints that are very much like marathons
It also:
- modernizes to golang 1.8.1
- replaces GoDeps dependencies of yester-year with golang vendoring
- updates etcd to 2.3.8 from 2.0.0 rc1
- paves the way for etcd 3
- provides a completely dockerized build like flight-director
- reduces the docker image size to 46M
- provides dev shell 
- add `make help` to see list of available commands

> Note: the bulk of this diff constitutes removing GoDeps/...*. See vendor/vendor.json

### CODE REVIEW

**Comment by iderdik** [comment]
Did you mean to check in the godeps directory?

File: scheduler/scheduler.go [hunk 1]
```diff
@@ -172,9 +176,32 @@ func NewEtcdScheduler(
...
 		reconciliationInfo:           map[string]string{},
+		constraints:                  Constraints{},
 	}
 }
 
+// AddRawConstraints -- parses a constraint as they'd be added by Marathon.
+func (s *EtcdScheduler) AddRawConstraints(statements string) error {
+	for _, tuple := range strings.Split(statements, ";") {
+		statement := strings.SplitN(tuple, ":", 3)
+		switch op := strings.ToUpper(statement[1]); op {
```
Comment (iderdik) [inline scheduler/scheduler.go:187]: What format is this code expecting the constraint to be in? Are we sure that the 2nd element of the array is always the operation?

Comment (f4tq) [reply scheduler/scheduler.go:187]: - Constraints marathon constraints LIKE, UNLIKE. 
- UNIQUE(enforced by existing framework switch), MAX_PER, CLUSTER, GROUP_BY are explicitly ignored 
- The entire marathon constraint group may be passed wholesale but only LIKE,UNLIKE are interpreted
- All constraints must pass just like marathon i.e assume AND not OR between constraints
- Just like Marathon, the form is:
    - noun:verb:optional*;
- See [go-related commit](https://github.com/adobe-platform/etcd-mesos/pull/1/commits/2507722c5b8f45e50405231758d75e8852a001ff) for golang constraint processing

- Everything else involves pulling out GoDeps/*
- git submodules
- Using go 1.8.1
- 2-phase build like FD, capcom, go-metronome, skopos, skms-gate etc
- Makefile to be  like FD, capcom, go-metronome, skopos, skms-gate etc

**Comment by f4tq** [comment]
- shouldn't we merge this now since we're using it in ethos01?

**Pull request merged**